docs/advanced-features/error-handling.md

```diff
@@ -0,0 +1,98 @@
+---
+description: Handle errors in your Next.js app.
+---
+
+# Error Handling
+
+This documentation explains how you can handle development, server-side, and client-side errors.
+
+## Handling Errors in Development
+
+When there is a runtime error during the development phase of your Next.js application, you will encounter an **overlay**. Fixing the error will automatically dismiss the overlay. It is only visible when the development server runs using `next dev`, `npm run dev`, or `yarn dev` and not in production.
+
+## Handling Server Errors
+
+Next.js provides a static 500 page by default to handle server-side errors that occur in your application. This page generates at build time. You can also [customize this page](/docs/advanced-features/custom-error-page#customizing-the-500-page) by creating a `pages/500.js` file.
+
+Having a 500 page in your application does not show specific errors to the app user.
```

Comment:
Should we also mention 404 pages (its basically a type of runtime error)

Reply:
Are 404 pages used to handle server errors? If yes, I can link the 404 pages doc here.

Reply:
Yes they handle a specific type of server error: file not found

https://nextjs.org/docs/advanced-features/custom-error-page#404-page